tokio/src/sync/watch.rs

```diff
@@ -551,14 +507,13 @@ impl<T> Receiver<T> {
     /// }
     /// ```
     pub fn has_changed(&self) -> Result<bool, error::RecvError> {
-        // Load the version from the state
-        let state = self.shared.state.load();
-        if state.is_closed() {
+        if self.shared.is_sender_closed() {
             // The sender has dropped.
             return Err(error::RecvError(()));
         }
-        let new_version = state.version();
 
+        // Load the version from the state
+        let new_version = self.shared.version();
         Ok(self.version != new_version)
     }
```

Comment:
This method seems to not match `maybe_changed` because the order of checking whether it's closed and the version has changed is different. The behavior in other places is that it will give one last `true` when all the Senders are gone before returning an error, but here it gives an error even when there's an unobserved change.

Maybe I should change it to match behavior?